### PR TITLE
some skill related mechanics + a few refactors

### PR DESCRIPTION
**Pull request by capturcus**
* added goldfish spawning
* added a mechanic where killing a goldfish gives you one of 4 skills at random: blink, mob attractor, mob dispersor, ink bomb
* implemented showing the skills on the client
* refactors

### CODE REVIEW

File: client/dfclient/src/gameplay_state.cpp [hunk 1]
```diff
@@ -161,7 +183,8 @@ void GameplayState::LoadLevel() {
 }
 
 // this whole thing should probably be refactored
```
Comment (mparafin) [inline client/dfclient/src/gameplay_state.cpp:185]: Is this not _the_ refactor commit? :D

File: common/constants.hpp [hunk 2]
```diff
@@ -24,4 +24,16 @@ const int CIV_SLOW_FRAMES = 40;
...
+
+enum class Skills {
+    INK_BOMB,
+    ATTRACTOR,
+    DISPERSOR,
+    BLINK,
+    SKILLS_MAX
+};
+
+const int MAX_SKILLS = 3;
```
Comment (mparafin) [inline common/constants.hpp]: Newline :P

File: server/mobs.cpp [hunk 3]
```diff
@@ -237,10 +237,26 @@ Mob::~Mob()
...
+
 void Civilian::handleKill(Player& killer) {
 	if (killer.isDead())
 		return;
 	this->toBeDeleted = true;
+	if (this->species == GOLDFISH_SPECIES) {
+		handleGoldfishKill(killer);
+		return;
+	}
 	killer.points += CIVILIAN_PENALTY;
```
Comment (mparafin) [inline server/mobs.cpp:260]: The civillian penalty shouldn't actually be imposed when killing a goldfish, I suppose

**Review by mparafin — CHANGES_REQUESTED** [review]
The game crashes randomly because of skill name inconsistencies, please fix xd Otherwise mergeable, beside some nitpicks. Good work :+1:

File: server/mobs.cpp (hunk 3 above)
Comment (capturcus) [reply server/mobs.cpp:260]: It isn't. The function returns before this line.

File: client/dfclient/src/gameplay_state.cpp [hunk 4]
```diff
@@ -63,13 +76,22 @@ std::unique_ptr<ncine::AnimatedSprite> GameplayState::CreateNewAnimSprite(ncine:
 		ret->addAnimation(nctl::move(animation));
 	}
 
-	ret->setAnimationIndex(FISH_ANIMATIONS::WALK);
+	ret->setAnimationIndex(0);
 	ret->setFrame(0);
 	ret->setPaused(false);
-	ret->setLayer(MOBS_LAYER);
+	ret->setLayer((unsigned short)Layers::MOBS);
```
Comment (piodul) [inline client/dfclient/src/gameplay_state.cpp]: I think that this should be passed through a parameter.

File: client/dfclient/src/gameplay_state.cpp [hunk 5]
```diff
@@ -384,6 +386,41 @@ void GameplayState::OnMessage(const std::string& data) {
 	this->currentHidingSpot = worldState->currentHidingSpot()->str();
 }
 
+void GameplayState::ProcessSkillBarUpdate(const void* ev) {
+	std::cout << "skill bar update received\n";
+	auto skillBarUpdate = (const FlatBuffGenerated::SkillBarUpdate*) ev;
+	this->skillIcons.resize(0);
```
Comment (piodul) [inline client/dfclient/src/gameplay_state.cpp]: `this->skillIcons.clear();` ?

File: client/dfclient/src/gameplay_state.cpp (hunk 1 above)
Comment (capturcus) [reply client/dfclient/src/gameplay_state.cpp:185]: This pr refactores _other things, okay?_

File: common/constants.hpp (hunk 2 above)
Comment (capturcus) [reply common/constants.hpp]: damn

File: client/dfclient/src/gameplay_state.cpp (hunk 4 above)
Comment (capturcus) [reply client/dfclient/src/gameplay_state.cpp]: And now it is.

File: client/dfclient/src/gameplay_state.cpp (hunk 5 above)
Comment (capturcus) [reply client/dfclient/src/gameplay_state.cpp]: oh well

**Review by mparafin — APPROVED** [review]
:+1:

**Pull request merged**